src/dotnet/commands/dotnet-run3/Run3Command.cs

```diff
@@ -87,30 +75,25 @@ private ITaskItem GetRunInformation()
 
             ProjectInstance projectInstance = new ProjectInstance(Project, globalProperties, null);
 
-            BuildRequestData buildRequestData = new BuildRequestData(projectInstance, new string[] { GetRunInformationTaskName });
-            BuildParameters buildParameters = new BuildParameters();
-
-            BuildResult result = BuildManager.DefaultBuildManager.Build(buildParameters, buildRequestData);
-
-            TargetResult runInfoResult;
-            if (!result.ResultsByTarget.TryGetValue(GetRunInformationTaskName, out runInfoResult))
+            string runProgram = projectInstance.GetPropertyValue(RunCommandPropName);
+            if (string.IsNullOrEmpty(runProgram))
             {
-                throw new InvalidOperationException($"Could not find a target named '{GetRunInformationTaskName}' in your project. Please ensure 'dotnet run' supports this project.");
+                throw new InvalidOperationException($"The property named '{RunCommandPropName}' does not have a value in your project. Please ensure 'dotnet run' supports this project.");
```

Comment:
Maybe I should check the `$(OutputType)` here and throw a decent error message saying that the $(OutputType) should be "Exe"?  Or not "Library"...  Or say "this is the output type: {OutputType}, ensure this is a runnable output type".

Thoughts?